code/datums/components/admin_popup.dm

```diff
@@ -0,0 +1,115 @@
+/// Applied to clients when they receive an admin popup, alerting them to
+/// their ticket.
+/datum/component/admin_popup
+	/// The user's most active ticket. If this is resolved, closed, or replied to,
+	/// then the component will delete itself.
+	var/datum/admin_help/ticket
+
+	var/atom/movable/screen/admin_popup/admin_popup
+
+/datum/component/admin_popup/Initialize(datum/admin_help/ticket)
+	if (!istype(parent, /client))
+		return COMPONENT_INCOMPATIBLE
+
+	if (!istype(ticket))
+		return COMPONENT_INCOMPATIBLE
+
+	create_notice()
+
+	RegisterSignal(ticket, COMSIG_ADMIN_HELP_INACTIVE, .proc/delete_self)
+	RegisterSignal(ticket, COMSIG_ADMIN_HELP_REPLIED, .proc/delete_self)
+	RegisterSignal(ticket, COMSIG_PARENT_QDELETING, .proc/delete_self)
```

Comment:
```suggestion
	RegisterSignal(
		ticket,
		list(
			COMSIG_ADMIN_HELP_INACTIVE,
			COMSIG_ADMIN_HELP_REPLIED,
			COMSIG_PARENT_QDELETING
			),
		.proc/delete_self,
		)
```
`RegisterSignal()` converts the signals into a list.
Feel free to format it differently if you think this is less readable.